### ISSUE TITLE
Error with constant cache for BAOAB

### ISSUE THREAD

**Issue by jebej** [SciML/StochasticDiffEq.jl#517]
The following errors out because the cache logic tries to `copyto!` to a `Float64`(I suppose it expected an array). I'm not too sure how the whole machinery works, so I don't know how to fix this.

```julia
using StochasticDiffEq
jj!(ϕ′,ϕ,τ,β,I) = (I - ϕ′ - sin(ϕ))/β
jj!(ϕ′,ϕ,p::Tuple,τ) = jj!(ϕ′,ϕ,τ,p[1],p[2])
identity_sde(v, u, p, t) = v # needed to form second order SDE
g(u,p,t) = 1

ϕ_init = 0.0
ϕ′_init = 0.0
tspan = (0.0,50.0)
β = 1.25
I = 2.0
prob = DynamicalSDEProblem{false}(jj!,identity_sde,g,ϕ′_init,ϕ_init,tspan,(β,I))

solve(prob,BAOAB(),dt=1/10)
```

**Comment by jebej** [comment]
Pinging @jamesgardner1421 who implemented this.

**Comment by ChrisRackauckas** [comment]
https://github.com/SciML/StochasticDiffEq.jl/pull/518 looks like a fine fix.

**Comment by jamesgardner1421** [comment]
Just opened a PR that should fix this for you. Originally I only ever tested it with arrays but it will also work with scalars once the PR is merged.

**Comment by jebej** [comment]
Thanks! I also wanted to ask about the form of the SDE, but I'll open a new issue.

**Comment by jebej** [comment]
After reading the paper a little bit, I now understand that the point of the BAOAB method is to include this term in the middle "O" operation for better accuracy. As a compromise for applications different than molecular sampling, I'd suggest going back to @jamesgardner1421's second proposition, where `gamma` is only included in the `gamma*v*dt` term, and not in the noise term. That way, one can choose `gamma=0` to disable this extra operation, or alternatively, it lets people choose exactly how to define the noise term without the current forced `gamma` dependence.

For my application, for example, where I need:
```
jj!(ϕ′,ϕ,τ,β,I) = (I - ϕ′ - sin(ϕ))/β
```
I can use
```
jj!(ϕ′,ϕ,τ,β,I) = (I - sin(ϕ))/β
```
with `gamma = 1/β`. However, I do not want the noise term to include this `β`.

I can make a PR, if that is acceptable.
